lib/spack/spack/test/config.py

```diff
@@ -1132,3 +1132,20 @@ def test_single_file_scope_cache_clearing(env_yaml):
     after = scope.get_section('config')
     assert after
     assert before == after
+
+
+@pytest.mark.regression('22611')
+def test_internal_config_scope_cache_clearing():
+    """
+    An InternalConfigScope object is constructed from data that is already
+    in memory, therefore it doesn't have any cache to clear. Here we ensure
+    that calling the clear method is a no-op.
+    """
+    config = spack.config.Configuration()
+    config.push_scope(spack.config.InternalConfigScope('internal', {
+        'config': {
+            'build_jobs': 10
+        }
+    }))
+    config.clear_caches()
+    assert config.get('config:build_jobs') == 10
```

Comment:
With respect to https://github.com/spack/spack/pull/22609/files#r603405923 what I am saying is that the bug doesn't need a `Configuration` object to be triggered. So something like:
```suggestion
@pytest.mark.regression('22611')
def test_internal_config_scope_cache_clearing():
    """
    An InternalConfigScope object is constructed from data that is already
    in memory, therefore it doesn't have any cache to clear. Here we ensure
    that calling the clear method is consistent with that..
    """
    data = {
        'config': {
            'build_jobs': 10
        }
    }
    internal_scope = spack.config.InternalConfigScope('internal', data)
    # Ensure that the initial object is properly set
    assert internal_scope.sections['config'] == data
    # Call the clear method
    internal_scope.clear()
    # Check that this didn't affect the scope
    assert internal_scope.sections['config'] == data
```
would be removing any interaction with `Configuration` and still testing the bug.